### PR TITLE
feature/287 warn when weaviate client outdated

### LINKED ISSUE #287: Warn when weaviate and/or client are outdated
Some users use very old versions of weaviate and client. We should warn them if it the minor version is 3 or more versions behind the last release.

Eg currently the last Weaviate version is 1.18.3, so we would warn users of 1.15 or below. We could do the same for the client versions.

### PR DESCRIPTION
**Pull request by iangregson**
Closes #287 :: warn when weaviate client outdated

* Adds a helper function in utils to compare current and latest version strings
* Fetches latest version from pypi when initialising a connection
* Compares latest pypi version with `__version__` and warns the user if more than 3 minor versions behind
* Don't fail if we can't get/parse any versions because that'll annoy the user
* Adds a new static warning method with appropriate message
* Adds some basic tests for the version comparison

### CODE REVIEW

**Comment by iangregson** [comment]
I agree ✅

File: test/test_util.py [hunk 1]
```diff
@@ -475,6 +476,21 @@ def test_parse_version_string(self):
         assert parse_version_string("2.0.0") == (2, 0)
         assert parse_version_string("v2.05.5") == (2, 5)
 
+    def test__is_weaviate_client_too_old(self):
```
Comment (dirkkul) [inline test/test_util.py:479]: could you use pytest+parametrization like in the test right below?

File: weaviate/warnings.py [hunk 2]
```diff
@@ -115,3 +115,12 @@ def weaviate_too_old_vs_latest(server_version: str):
             category=DeprecationWarning,
             stacklevel=1,
         )
+
+    @staticmethod
+    def weaviate_client_too_old_vs_latest(client_version: str, latest_version: str):
+        warnings.warn(
+            message=f"""Dep004: You are using weaviate-client version {client_version}. The latest version is {latest_version}.
```
Comment (dirkkul) [inline weaviate/warnings.py:122]: Dep005

**Review by dirkkul — COMMENTED** [review]
Thank you for your contribution! There are some minor issues, would be great if you could fix those. In addition to the comments it would also be great if you could run black to fix the linter issues.

Sorry that my review took so long, I was on holiday :)

File: weaviate/warnings.py (hunk 2 above)
Comment (iangregson) [reply weaviate/warnings.py:122]: Hi Dirk! Sure thing.

Also apologies for delay - also had vacation :)

I'll get these changes in this week.

Comment (dirkkul) [reply weaviate/warnings.py:122]: thank you very much!

**Pull request merged**